### PR TITLE
230 basic linting rule f

### PR DESCRIPTION
**Pull request by mschwoer**
-    add rule "F" to ruleset
-    make code pass rule "F" (see individual commits)
@GeorgWa please check carefull all TODOs introduced in here, they might be potential bugs. If not, I'll remove the unused variables in a subsequent PR

### CODE REVIEW

File: alphadia/outputtransform.py [hunk 1]
```diff
@@ -402,9 +400,9 @@ def build_transfer_model(self):
         tune_mgr = FinetuneManager(
             device=device, settings=self.config["transfer_learning"]
         )
-        stats = tune_mgr.finetune_rt(transfer_lib.precursor_df)
-        stats = tune_mgr.finetune_charge(transfer_lib.precursor_df)
-        stats = tune_mgr.finetune_ms2(
+        stats = tune_mgr.finetune_rt(transfer_lib.precursor_df)  # noqa: F841  # TODO check for potential bug then remove this line
```
Comment (GeorgWa) [inline alphadia/outputtransform.py:403]: These lines have to stay. Kinda weird, that ruff want's to remove the function calls.

File: tests/unit_tests/test_fdr.py [hunk 2]
```diff
@@ -258,11 +258,11 @@ def test_feed_forward():
     # assert classifier.metrics["test_accuracy"][-1] > 0.99
     # assert classifier.metrics["train_accuracy"][-1] > 0.99
 
-    y_pred = classifier.predict(x)
-    # assert np.all(y_pred == y)
+    y_pred = classifier.predict(x)  # noqa: F841  # TODO check for potential bug then remove this line
```
Comment (GeorgWa) [inline tests/unit_tests/test_fdr.py:261]: All of the calls in this file have to stay

**Review by GeorgWa — APPROVED** [review]
LGTM,
All changes can be applied apart from the two marked files.

File: alphadia/outputtransform.py (hunk 1 above)
Comment (mschwoer) [reply alphadia/outputtransform.py:403]: ruff wants to remove the unused variables, and is not aware of the called methods altering some internal state in `tune_mgr` .. 
ruff is happy if the variable name is changed to "_", i.e.
```_ = tune_mgr. ...```
or if the variable assignment is removed .. I'll do the latter ..

Comment (mschwoer) [reply alphadia/outputtransform.py:403]: => https://github.com/MannLabs/alphadia/pull/236

**Pull request merged**